devlib/platform/arm.py

```diff
@@ -292,3 +294,162 @@ def take_measurement(self):
+        self.gem5_port = gem5_telnet_port or -1
+        self.stats_directory = host_output_dir
+        self.gem5outdir = os.path.join(self.stats_directory, "gem5")
+        self.gem5interactdir = '/tmp'
+        self.executable_dir = None
+        self.working_dir = None
+        self.m5_path = 'm5'
+        self.stdout_file = None
+        self.stderr_file = None
+        self.stderr_filename = None
+        if self.gem5_port == -1:
+            #Allows devlib to pick up already running simulations
+            self.start_gem5_simulation = True
+        else:
+            self.start_gem5_simulation = False
+
+        # Find the first one that does not exist. Ensures that we do not re-use
+        # the directory used by someone else.
+        for i in xrange(sys.maxint):
+            directory = os.path.join(self.gem5interactdir, "wa_{}".format(i))
+            try:
+                os.stat(directory)
+                continue
+            except OSError:
+                break
+        self.gem5interactdir = directory
+        self.logger.debug("Using {} as the temporary directory.".format(self.gem5interactdir))
+
+        # Parameters passed onto gem5
+        self.gem5args_binary = gem5_bin
+        self.gem5args_args = gem5_args
+        self.gem5args_virtio = gem5_virtio
+        self._check_gem5_command()
+
+        # Start the interaction with gem5
+        self._start_interaction_gem5()
+
+    def _check_gem5_command(self):
+        """
+        Check if the command to start gem5 makes sense
+        """
+        assert(self.gem5args_binary is not None)
+        assert(self.gem5args_args is not None)
+        self.gem5args_virtio = str(self.gem5args_virtio).format(self.gem5interactdir)
+        assert(self.gem5args_virtio is not None)
+
+
+    def _start_interaction_gem5(self):
+        """
+        Starts the interaction of devlib with gem5.
+        """
+
+        # First create the input and output directories for gem5
+        if self.start_gem5_simulation:
+            self.logger.info("Creating temporary directory: {}".format(self.gem5interactdir))
+            if not os.path.exists(self.stats_directory):
+                os.mkdir(self.stats_directory)
+            os.mkdir(self.gem5interactdir)
+            if os.path.exists(self.gem5outdir):
+               self.logger.debug("The output directory {} already existed." \
+                                 " Previous results will be overwritten.".format(self.gem5outdir))
+            else:
+                os.mkdir(self.gem5outdir)
+
+            # We need to redirect the standard output and standard error for the
+            # gem5 process to a file so that we can debug when things go wrong.
+            f = os.path.join(self.gem5outdir, 'stdout')
+            self.stdout_file = open(f, 'w')
+            f = os.path.join(self.gem5outdir, 'stderr')
+            self.stderr_file = open(f, 'w')
+            # We need to keep this so we can check which port to use for the telnet
```

Comment:
I don't think that works here as you open these files here and then pass them on to the gem5 subprocess, which is the one that will actually write to these files.